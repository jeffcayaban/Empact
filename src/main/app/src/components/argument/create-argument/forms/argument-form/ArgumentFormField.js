import React from "react";
import { ControlLabel, FormControl, FormGroup } from "react-bootstrap";

/**
 * Represents a generic field for the argument form. It indicates the character count for the field.
 */

class ArgumentFormField extends React.PureComponent {
    render() {
        const { id, label, maxLength, value,  ...props } = this.props;
        const currentFieldLength = value !== undefined ? value.length : 0;
        return (
            <FormGroup controlId={id} style={{ paddingBottom: '10px' }}>
                <ControlLabel>{label}</ControlLabel>
                <FormControl maxLength={maxLength} value={value} {...props} />
                <FormControl.Feedback />
                <div className={"argFormFieldCharCount"}>
                    <div className={"titleLengthCounter"}>
                        <p>{currentFieldLength}/{maxLength}</p>
                    </div>
                </div>
            </FormGroup>
        );
    }
}

export default ArgumentFormField;