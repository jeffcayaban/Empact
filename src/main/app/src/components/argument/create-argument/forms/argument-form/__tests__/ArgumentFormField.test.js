import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import ArgumentFormField from "../ArgumentFormField";

describe('ArgumentFormField', () => {

    it('should render successfully', () => {
        const component = shallow(
            <ArgumentFormField
                id={"testId"}
                label={"testLabel"}
                maxLength={90}
                value={"testValue"}
            />
        );

        expect(toJson(component)).toMatchSnapshot();
    });
});
