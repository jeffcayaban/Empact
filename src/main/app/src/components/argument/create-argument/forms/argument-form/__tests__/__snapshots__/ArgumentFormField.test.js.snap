// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArgumentFormField should render successfully 1`] = `
<FormGroup
  bsClass="form-group"
  controlId="testId"
  style={
    Object {
      "paddingBottom": "10px",
    }
  }
>
  <ControlLabel
    bsClass="control-label"
    srOnly={false}
  >
    testLabel
  </ControlLabel>
  <FormControl
    bsClass="form-control"
    componentClass="input"
    maxLength={90}
    value="testValue"
  />
  <FormControlFeedback
    bsClass="form-control-feedback"
    bsRole="feedback"
  />
  <div
    className="argFormFieldCharCount"
  >
    <div
      className="titleLengthCounter"
    >
      <p>
        9
        /
        90
      </p>
    </div>
  </div>
</FormGroup>
`;
